vizro-core/examples/visual-vocabulary/tools/generate_vivivo_json.py

```diff
@@ -0,0 +1,114 @@
+"""Generate visual vocabulary JSON from vivivo directory."""
+
+import importlib
+import json
+import re
+import sys
+from pathlib import Path
+
+# Add parent directory to path so we can import chart_groups
+sys.path.append(str(Path(__file__).resolve().parent.parent))
+from chart_groups import CHART_GROUPS
+
+
+def extract_card_text_sections(text):
+    """Extract 'What is' and 'When should I use it' sections from card text."""
+    what_pattern = r"#### What is .*?\?\s+(.*?)(?=####|$)"
+    when_pattern = r"#### When should I use it\??\s+(.*?)(?=$)"
+
+    what_match = re.search(what_pattern, text, re.DOTALL)
+    when_match = re.search(when_pattern, text, re.DOTALL)
+
+    what_text = what_match.group(1).strip() if what_match else ""
+    when_text = when_match.group(1).strip() if when_match else ""
+
+    # Clean up HTML elements and trailing non-breaking spaces
+    what_text = re.sub(r"\n\s*&nbsp;\s*$", "", what_text)
+
+    return what_text, when_text
+
+
+def get_example_code(var_name, group_name=None):
+    """Get the example code for a chart type."""
+    script_dir = Path(__file__).resolve().parent
+    examples_dir = script_dir.parent / "pages" / "examples"
+
+    possible_files = [
+        f"{group_name}_{var_name}.py",  # e.g., time_column.py
+        f"{var_name}.py",  # e.g., column.py
+    ]
+
+    for filename in possible_files:
+        example_file = examples_dir / filename
+        if example_file.exists():
+            with open(example_file) as f:
+                return f.read()
+
+    return ""
+
+
+def get_chart_name_from_example(var_name, group_name=None):
+    """Get the chart name from the example file."""
+    if group_name and var_name.startswith(f"{group_name}_"):
+        return var_name[len(group_name) + 1 :]
+    return var_name
+
+
+def generate_visual_vocabulary():
```

Comment:
I think this function could do with a few more comments to make it clear what it's doing and what its assumptions are. e.g. if we change the layout of the ViViVo in future, which bit here is likely to break? We assume a certain structure and that Card is used to hold the content, etc.